base/linalg/matmul.jl

```diff
@@ -656,6 +656,7 @@ function matmul2x2{T,S}(tA, tB, A::AbstractMatrix{T}, B::AbstractMatrix{S})
 end
 
 function matmul2x2!{T,S,R}(C::AbstractMatrix{R}, tA, tB, A::AbstractMatrix{T}, B::AbstractMatrix{S})
+    size(C) == (2, 2) || throw(DimensionMismatch("Destination matrix is not 2×2"))
```

Comment:
In my opinion, this error message would be better to look identical to the one from the larger matrix case.